File: dataModule/src/main/java/com/mobnetic/coinguardian/util/CurrencyUtils.java
package com.mobnetic.coinguardian.util;

import com.mobnetic.coinguardian.model.CurrencySubunit;
import com.mobnetic.coinguardian.model.CurrencySubunitsMap;
import com.mobnetic.coinguardian.model.currency.CurrenciesSubunits;
import com.mobnetic.coinguardian.model.currency.CurrencySymbols;

public class CurrencyUtils {

	public static String getCurrencySymbol(String currency) {
		return CurrencySymbols.CURRENCY_SYMBOLS.containsKey(currency) ? CurrencySymbols.CURRENCY_SYMBOLS.get(currency) : currency;
	}
	
	public static CurrencySubunit getCurrencySubunit(String currency, long subunitToUnit) {
		if(CurrenciesSubunits.CURRENCIES_SUBUNITS.containsKey(currency)) {
			CurrencySubunitsMap subunits = CurrenciesSubunits.CURRENCIES_SUBUNITS.get(currency);
			if(subunits.containsKey(subunitToUnit))
				return subunits.get(subunitToUnit);
		}
		
		return new CurrencySubunit(currency, 1);
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/util/CurrencyUtils.kt
package com.aneonex.bitcoinchecker.datamodule.util

import com.aneonex.bitcoinchecker.datamodule.model.CurrencySubunit
import com.aneonex.bitcoinchecker.datamodule.model.currency.CurrenciesSubunits
import com.aneonex.bitcoinchecker.datamodule.model.currency.CurrencySymbols

object CurrencyUtils {
    fun getCurrencySymbol(currency: String): String {
        return CurrencySymbols.CURRENCY_SYMBOLS[currency] ?: currency
    }

    fun getCurrencySubunit(currency: String?, subunitToUnit: Long): CurrencySubunit {
        val subunits = CurrenciesSubunits.CURRENCIES_SUBUNITS[currency]
        if (subunits != null) {
            if (subunits.containsKey(subunitToUnit)) return subunits.getValue(subunitToUnit)
        }
        return CurrencySubunit(currency ?: "", 1)
    }
}